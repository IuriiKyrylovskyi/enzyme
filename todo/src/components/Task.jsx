import React from "react";
import { PropTypes } from "prop-types";

const Task = ({ id, done, text, handleStatusChange, handleTaskDelete }) => {
  const listItemClass = `list-item ${done ? "list-item_done" : ""}`;

  return (
    <li className={listItemClass}>
      <input
        type="checkbox"
        className="list-item__checkbox"
        checked={done}
        onChange={() => {
          handleStatusChange(id);
        }}

        //
      />
      {text}
      <button
        className="list-item__delete-btn"
        onClick={() => {
          handleTaskDelete(id);
        }}
        //
      />
    </li>
  );
};

Task.propTypes = {
  id: PropTypes.string.isRequired,
  done: PropTypes.bool,
  text: PropTypes.string,
  handleStatusChange: PropTypes.func.isRequired,
  handleTaskDelete: PropTypes.func.isRequired,
};

// Task.defaultProps = {
//   done: false,
//   text: "",
// };

export default Task;
