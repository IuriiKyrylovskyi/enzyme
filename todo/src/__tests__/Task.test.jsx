import React from "react";
import { shallow } from "enzyme";
import Task from "../components/Task";

describe("<Task />", () => {
  it("should display done task", () => {
    const props = {
      id: "mock-id",
      text: "mock-text",
      done: true,
      handleStatusChange: jest.fn(),
      handleTaskDelete: jest.fn(),
    };
    const wrappedComponent = shallow(<Task {...props} />);

    expect(wrappedComponent).toMatchSnapshot();
  });

  it("should display undone task", () => {
    const props = {
      id: "mock-id-2",
      text: "Task 2",
      done: false,
      handleStatusChange: jest.fn(),
      handleTaskDelete: jest.fn(),
    };
    const wrappedComponent = shallow(<Task {...props} />);

    expect(wrappedComponent).toMatchSnapshot();
  });

  it("should update task on checkbox checked", () => {
    const props = {
      id: "mock-id-3",
      text: "Task 3",
      done: false,
      handleStatusChange: jest.fn(),
      handleTaskDelete: jest.fn(),
    };
    const wrappedComponent = shallow(<Task {...props} />);
    wrappedComponent.find(".list-item__checkbox").simulate("change");
    
    expect(props.handleStatusChange).toBeCalledWith("mock-id-3");
  });

  it("should delete task", () => {
    const props = {
      id: "mock-id-4",
      text: "Task 4",
      done: true,
      handleStatusChange: jest.fn(),
      handleTaskDelete: jest.fn(),
    };
    const wrappedComponent = shallow(<Task {...props} />);
    wrappedComponent.find(".list-item__delete-btn").simulate("click");

    expect(props.handleTaskDelete).toBeCalledWith("mock-id-4");
  });
});
