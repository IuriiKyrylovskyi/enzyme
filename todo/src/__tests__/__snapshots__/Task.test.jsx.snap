// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> should display done task 1`] = `
<li
  className="list-item list-item_done"
>
  <input
    checked={true}
    className="list-item__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  mock-text
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;

exports[`<Task /> should display undone task 1`] = `
<li
  className="list-item "
>
  <input
    checked={false}
    className="list-item__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  Task 2
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;
